### PR TITLE
Bump pip-tools from 5.4.0 to 5.5.0 in /python/helpers and update test spec

### PR DESCRIPTION
**Pull request by relud**
As of pip-tools 5.5, the output format changed to always put `via` comments on separate lines, and no longer include a trailing `\` when using `--generate-hashes`.

so it changed from this output:
```
#
# This file is autogenerated by pip-compile
# To update, run:
#
#    pip-compile --output-file requirements/test.txt requirements/test.in
#
apipkg==1.4               # via execnet
attrs==18.1.0             # via -r requirements/test.in, pytest
execnet==1.5.0            # via pytest-xdist
flaky==3.4.0              # via -r requirements/test.in
mock==2.0.0               # via -r requirements/test.in
more-itertools==4.1.0     # via pytest
pbr==4.0.2                # via mock
pluggy==0.6.0             # via pytest
py==1.5.3                 # via pytest
pytest-forked==0.2        # via pytest-xdist
pytest-xdist==1.22.2      # via -r requirements/test.in
pytest==3.5.1             # via -r requirements/test.in, pytest-forked, pytest-xdist
six==1.11.0               # via mock, more-itertools, pytest
```
to this output:
```
#
# This file is autogenerated by pip-compile
# To update, run:
#
#    pip-compile --output-file requirements/test.txt requirements/test.in
#
apipkg==1.4
    # via execnet
attrs==20.3.0
    # via
    #   -r requirements/test.in
    #   pytest
execnet==1.5.0
    # via pytest-xdist
flaky==3.4.0
    # via -r requirements/test.in
mock==2.0.0
    # via -r requirements/test.in
more-itertools==4.1.0
    # via pytest
pbr==4.2.0
    # via mock
pluggy==0.6.0
    # via pytest
py==1.5.3
    # via pytest
pytest-forked==0.2
    # via pytest-xdist
pytest-xdist==1.22.2
    # via -r requirements/test.in
pytest==3.5.1
    # via
    #   -r requirements/test.in
    #   pytest-forked
    #   pytest-xdist
six==1.11.0
    # via
    #   mock
    #   more-itertools
    #   pytest
```

### CODE REVIEW

**Comment by feelepxyz** [comment]
@relud thanks for updating the tests. Looks like this change is breaking so not sure what the best course of action is yet. Also seems breaking with the current version if you start using the new version so wondering if the best approach would be install both versions and switch based on the formatting.

Ref: https://github.com/jazzband/pip-tools/pull/1237#issuecomment-754514296

**Comment by jurre** [comment]
As mentioned in the issue @feelepxyz linked to, I wouldn't feel terrible for just merging this as the conflicts are merely cosmetic. There shouldn't be anything stopping people from upgrading their own version of pip-tools, and in the worst case our diffs will be a little noisy.

**Comment by feelepxyz** [comment]
Yeah agreed, I think we should go ahead and merge this. Will have a look over and test tomorrow.

**Comment by dannyshaw** [comment]
@feelepxyz just replying from your question in https://github.com/jazzband/pip-tools/pull/1237#issuecomment-754514296

Considering the rendered output of pip-compile cannot be relied on, I wondered if an approach of installing the version of pip-tools referenced in the requirements.in file itself (if exists) and using that version to compile the requirements may be an option?

**Review by feelepxyz — APPROVED** [review]
LGTM 👍  Seems safe to not treat this as a breaking change and accept some potential for noisy diffs until everyone has upgraded.

**Comment by feelepxyz** [comment]
> Considering the rendered output of pip-compile cannot be relied on, I wondered if an approach of installing the version of pip-tools referenced in the project's requirements.in file itself (if it exists) and using that version to compile the requirements may be an option?

Yeah good suggestion, not sure it's worth the effort to parse and install a custom version based on the requirements.in though, seems ok to not treat this change as breaking following the sentiment from the pip-tools folks.

**Comment by feelepxyz** [comment]
This will probably generate some support volume as the first PR with 5.5 will re-write the entire generated requirements file creating a potentially large and hard to parse diff.

We could offset this by including a message in the pull request description about the upgrade to 5.5 but not entirely sure how to do this yet. Will have a quick look at this.

Checking the first 10 pages of open public dependabot PRs not many are using pip-compile, I only found one repo + forks of it: https://github.com/SFDO-Tooling/Metecho/pull/408/files

Source: https://github.com/pulls?q=is%3Apr+author%3Aapp%2Fdependabot+label%3Apython+is%3Aopen

**Comment by jerr0328** [comment]
Hi, just want to give some feedback from someone using pip-tools in my repos (and private repos), since we have pip-tools in our requirements-dev.in file we saw the release notes about the changed requirements.txt format. I understand there's not really an easy way to support a smooth migration, but I think the "best" approach for these is to use the newer version, as the newer version's format makes diffs a bit better (after the first reformat) and Dependabot opens the pip-tools PR for us anyways.

If there are some confused people, perhaps a pinned issue can help mitigate some of the support traffic?

**Comment by cancan101** [comment]
@feelepxyz, why did this get reverted?

**Comment by feelepxyz** [comment]
@cancan101 yeah sorry about this, we're planning to release on Tuesday following some delays getting docs published on github.

**Comment by jerr0328** [comment]
The new PR is #2985 (to help linking these PRs and issues)
